projects/igniteui-angular/src/lib/directives/dragdrop/dragdrop.directive.ts

```diff
@@ -62,60 +62,181 @@ export class IgxDropEventArgs {
 })
 export class IgxDragDirective implements OnInit, OnDestroy {
 
+    /**
+     * @hidden
+     */
     @Input('igxDrag')
     public data: any;
 
+    /**
+     * An @Input property that indicates when the drag should start
+     * By default the drag starts after the draggable element is moved by 5px
+     * ```html
+     * <div igxDrag [dragTolerance]="100">
+     *         <span>Drag Me!</span>
+     * </div>
+     * ```
+     */
     @Input()
     public dragTolerance = 5;
 
+    /**
+     * @hidden
+     */
     @Input()
     public ghostImageClass = '';
 
+    /**
+     * An @Input property that hides the draggable element.
+     * By default it's set to false.
+     * ```html
+     * <div igxDrag [dragTolerance]="100" [hideBaseOnDrag]="'true'">
+     *         <span>Drag Me!</span>
+     * </div>
+     * ```
+     */
     @Input()
     public hideBaseOnDrag = false;
 
+    /**
+     * An @Input property that enables/disables the draggable element animation
+     * when the element is released.
+     * By default it's set to false.
+     * ```html
+     * <div igxDrag [animateOnRelease]="'true'">
+     *         <span>Drag Me!</span>
+     * </div>
+     * ```
+     */
     @Input()
     public animateOnRelease = false;
 
-    @Input()
-    public zIndexDrag = 20;
-
+    /**
+     * Emits an event when the draggable element drag starts.
+     * ```html
+     * <div igxDrag [animateOnRelease]="'true'" (dragStart)="onDragStart()">
+     *         <span>Drag Me!</span>
+     * </div>
+     * ```
+     * ```typescript
+     * public onDragStart(){
+     *      alert("The drag has stared!");
+     * }
+     * ```
+     */
     @Output()
     public dragStart = new EventEmitter<any>();
 
+    /**
+     * Emits an event when the draggable element drag ends.
+     * ```html
+     * <div igxDrag [animateOnRelease]="'true'" (dragEnd)="onDragEnd()">
+     *         <span>Drag Me!</span>
+     * </div>
+     * ```
+     * ```typescript
+     * public onDragEnd(){
+     *      alert("The drag has ended!");
+     * }
+     * ```
+     */
     @Output()
     public dragEnd = new EventEmitter<any>();
 
+    /**
+     * Emits an event when the draggable element move ends.
```

Comment:
I would say more like `Event triggered after the draggable element is released and after its animation has finished`